Packet++/header/IcmpV6Layer.h

```diff
@@ -0,0 +1,300 @@
+	/** Multicast Listener Report Message */
+	ICMPv6_MULTICAST_LISTENER_DISCOVERY_REPORTS = 143,
+	/** Home Agent Address Discovery Request Message */
+	ICMPv6_HOME_AGENT_ADDRESS_DISCOVERY_REQUEST_MESSAGE = 144,
+	/** Home Agent Address Discovery Reply Message */
+	ICMPv6_HOME_AGENT_ADDRESS_DISCOVERY_REPLY_MESSAGE = 145,
+	/** Mobile Prefix Solicitation Message */
+	ICMPv6_MOBILE_PREFIX_SOLICITATION = 146,
+	/** Mobile Prefix Advertisement Message */
+	ICMPv6_MOBILE_PREFIX_ADVERTISEMENT = 147,
+	/** Certification Path Solicitation Message */
+	ICMPv6_CERTIFICATION_PATH_SOLICITATION = 148,
+	/** Certification Path Advertisement Message */
+	ICMPv6_CERTIFICATION_PATH_ADVERTISEMENT = 149,
+	/** ICMP Experimental Mobility Subtype Format and Registry Message */
+	ICMPv6_EXPERIMENTAL_MOBILITY = 150,
+	/** Multicast Router Advertisement Message */
+	ICMPv6_MULTICAST_ROUTER_ADVERTISEMENT = 151,
+	/** Multicast Router Solicitation Message */
+	ICMPv6_MULTICAST_ROUTER_SOLICITATION = 152,
+	/** Multicast Router Termination Message*/
+	ICMPv6_MULTICAST_ROUTER_TERMINATION = 153,
+	/** RPL Control Message */
+	ICMPv6_RPL_CONTROL_MESSAGE = 155,
+	/** Private Experimentation Message */
+	ICMPv6_PRIVATE_EXPERIMENTATION3 = 200,
+	/** Private Experimentation Message */
+	ICMPv6_PRIVATE_EXPERIMENTATION4 = 201,
+	/** Reserved for expansion of ICMPv6 informational messages */
+	ICMPv6_RESERVED_EXPANSION_INFORMATIONAL = 255
+};
+
+/**
+ * @struct icmpv6hdr
+ * Represents an ICMPv6 protocol header
+ */
+#pragma pack(push, 1)
+struct icmpv6hdr
+{
+	/** Type of the message. Values in the range from 0 to 127 (high-order bit is 0) indicate an error message,
+	while values in the range from 128 to 255 (high-order bit is 1) indicate an information message. */
+	uint8_t type;
+	/** The code field value depends on the message type and provides an additional level of message granularity */
+	uint8_t code;
+	/** The checksum field provides a minimal level of integrity verification for the ICMP message */
+	uint16_t checksum;
+};
+#pragma pack(pop)
+
+/**
+ * @struct icmpv6_echo_hdr
+ * ICMP echo request/reply message structure
+ */
+#pragma pack(push, 1)
+typedef struct icmpv6_echo_hdr : icmpv6hdr
+{
+	/** the echo request identifier */
+	uint16_t id;
+	/** the echo request sequence number */
+	uint16_t sequence;
+} icmpv6_echo_hdr;
+#pragma pack(pop)
+
+/**
+ * @struct icmpv6_echo_request
+ * ICMPv6 echo request/reply message structure
+ */
+typedef struct icmpv6_echo_request
+{
+	/** a pointer to the header data */
+	icmpv6_echo_hdr *header;
+	/** most echo requests/replies contain some payload data. This is the data length */
+	size_t dataLength;
+	/** most echo requests/replies contain some payload data. This is a pointer to this data */
+	uint8_t *data;
+} icmpv6_echo_request;
+
+/**
+ * @typedef icmpv6_echo_reply
+ * ICMPv6 echo reply message structure, same as icmpv6_echo_request
+ */
+typedef icmpv6_echo_request icmpv6_echo_reply;

```

Comment:
I'm not sure we need this struct, now that we have dedicated layers for the echo/reply messages. The layer classes can return the header, the data length and a pointer to the data